### PR TITLE
Only test _is_param if doing instance check on Parameter base

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130578

Fixes https://github.com/pytorch/pytorch/issues/111348

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

File: torch/nn/parameter.py [hunk 1]
```diff
@@ -8,9 +8,12 @@
 class _ParameterMeta(torch._C._TensorMeta):
     # Make `isinstance(t, Parameter)` return True for custom tensor instances that have the _is_param flag.
     def __instancecheck__(self, instance):
-        return super().__instancecheck__(instance) or (
-            isinstance(instance, torch.Tensor) and getattr(instance, "_is_param", False)
-        )
+        if self is Parameter:
+            if isinstance(instance, torch.Tensor) and getattr(
```
Comment (Skylion007) [inline torch/nn/parameter.py:12]: nit can be one if statement, also can technically be one boolean although this might be marginally more readable.

**Review by Skylion007 — APPROVED** [review]

File: torch/nn/parameter.py (hunk 1 above)
Comment (ezyang) [reply torch/nn/parameter.py:12]: yeah, I know this can be golfed but this seemed like good code flow to me

**Comment by will-cromar** [comment]
Thanks!

**Pull request closed without merging**